public/app/plugins/panel/canvas/CanvasPanel.tsx

```diff
@@ -43,7 +42,19 @@ export class CanvasPanel extends Component<Props, State> {
   }
 
   componentDidMount() {
-    theScene.next(this.scene);
+    lastLoadedScene = this.scene;
+    if (window.location.href.indexOf('editPanel=') > 0) {
```

Comment:
Is there a better way to know if we are in edit mode?

Reply:
If this was a functional component you could have used `useUrlParams` hook:
```
const [params, updateUrlParams] = useUrlParams();
```

You can try wrapping CanvasPanel with `withRouter` HOC, to get the search params, but I think both approaches is an overkill. 

I agree with @torkelo that isInEditMode / isEditing should be just passed via props to any panel.